docs/ibc/light-clients/client-state.md

```diff
@@ -1,3 +1,65 @@
 <!--
 order: 2
--->
\ No newline at end of file
+-->
+
+# Implementing the ClientState interface
+
+Learn how to implement the [Client State](https://github.com/cosmos/ibc-go/blob/main/modules/core/exported/client.go#L36) interface.
+
+### ClientType
+
+`ClientType` should return a unique string identifier of the light client.
+
+### GetLatestHeight
+
+`GetLatestHeight` should return the latest block height.
+
+### Validate
+
+`Validate` should validate every client state field and should return an error if any value is invalid. The light client
+implementor is in charge of determining which checks are required. See the [tendermint light client implementation](https://github.com/cosmos/ibc-go/blob/main/modules/light-clients/07-tendermint/client_state.go#L111)
+as a reference.
+
+### Status
+
+`Status` must return the status of the client. Only `Active` clients are allowed to process packets. All
+possible Status types can be found [here](https://github.com/cosmos/ibc-go/blob/main/modules/core/exported/client.go).
+
+### ZeroCustomFields
+
+`ZeroCustomFields` should return a copy of the light client with all client customizable fields with their zero value. It should not mutate the fields of the light client.
+This method is used to [verify upgrades](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/types/proposal.go#L120) and when [scheduling upgrades](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/keeper/proposal.go#L82).
+
+### GetTimestampAtHeight
+
+`GetTimestampAtHeight` must return the timestamp for the consensus state associated with the provided height.
+
+### Initialize
+
+Clients must validate the initial consensus state, and may store any client-specific metadata necessary
+for correct light client operations in the `Initialize` function.
+
+`Initialize` gets called when a [client is created](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/keeper/client.go#L32).
+
+### VerifyMembership
+
+`VerifyMembership` must verify the existence of a value at a given CommitmentPath at the specified height.
+The caller of this function is expected to construct the full CommitmentPath from a CommitmentPrefix and a standardized
+path (as defined in ICS 24).
```

Comment:
Agree, but I think we've added a page for Membership/Non-Membership proofs here: https://github.com/cosmos/ibc-go/blob/main/docs/.vuepress/config.js#L316

Maybe the specific information could go under that such as CommitmentPath and CommitmentPrefix, explaining what they mean wrt ics24..etc. And then we link to it from here and underneath in `VerifyNonMembership` section also? 

We could then be concise here and just briefly mention that these methods are used for verification of existence and absence proofs of counterparty state using the `ClientState` created on-chain.

This is just a suggestion, interested to hear what others think regarding how this should be laid out, but as @colin-axner mentioned it will likely grow organically as we progress!

Reply:
ah good point @damiannolan, I think it makes sense to link to the proofs page and give the short version here.

Reply:
sounds good to me!